lib/alerts_viewer_web/live/bus_live.ex

```diff
@@ -18,7 +18,13 @@ defmodule AlertsViewerWeb.BusLive do
     current_algorithm = hd(delay_alert_algorithm_components)
 
     bus_routes = Routes.all_bus_routes()
-    bus_alerts = if(connected?(socket), do: Alerts.subscribe() |> filtered_by_bus(), else: [])
+
+    bus_alerts =
+      if(connected?(socket),
+        do: Alerts.subscribe() |> filtered_by_bus() |> filtered_by_delay_type(),
```

Comment:
suggestion: I believe that there's one additional place where we are filtering for bus alerts and that we also need to add `filtered_by_delay_type` to. That's when we get new alerts coming in down in `handle_info({:alerts, alerts}`.

Reply:
good catch!